docs/api/mcp/index.mdx

```diff
@@ -0,0 +1,273 @@
+# Sourcegraph MCP Server
+
+<p className="subtitle">Connect AI agents and applications to your Sourcegraph instance's code search and analysis capabilities.</p>
+
+The Sourcegraph Model Context Protocol (MCP) Server provides AI agents and applications with programmatic access to your Sourcegraph instance's code search, navigation, and analysis capabilities through a standardized interface.
+
+<Callout type="info">MCP Server requires Sourcegraph 6.8+ and is available on all deployment types.</Callout>
+
+## Server Endpoint
+
+The MCP server is available at:
+
+```
+https://your-sourcegraph-instance.com/.api/mcp/v1
+```
+
+## Authentication
+
+The Sourcegraph MCP server supports three authentication methods:
+
+### Access Token (Query Parameter)
+
+Pass your access token as a query parameter:
+
+```
+https://your-sourcegraph-instance.com/.api/mcp/v1?token=YOUR_ACCESS_TOKEN
+```
+
+### Authorization Header
+
+Include your token in the Authorization header:
+
+```
+Authorization: token YOUR_ACCESS_TOKEN
+```
+
+### OAuth 2.0 Device Flow
```

Comment:
Hmm, I'm not sure device flow is required here, that just depends on the MCP client -- auth code flow with PKCE can also be fine. So maybe just call it OAuth 2.0?

Also, let's make OAuth 2.0 the first option in the list, as it's the most spec-compliant and most secure option